pkg/gatewayserver/gatewayserver.go

```diff
@@ -894,19 +896,23 @@ func (gs *GatewayServer) handleUpstream(ctx context.Context, conn connectionEntr
 			correlationIDs = append(correlationIDs, fmt.Sprintf("gs:uplink:%s", events.NewCorrelationID()))
 			ctx = events.ContextWithCorrelationID(ctx, correlationIDs...)
 			msg.Message.CorrelationIds = events.CorrelationIDsFromContext(ctx)
+			registerReceiveUplink(ctx, gtw, msg, protocol)
```

Comment:
The subscribers (NOC and Console) really do depend on the payload being there in the event.

The whole point of the `UnmarshalMessage` done below is to allow the payload to be visible. If we emit the `gs.up.receive` before that, there is no event that actually shows the payload - `receive` has only the binary payload, `drop` has the error and `forward` has the host.

I think that it is better to have a useful `receive` than to move it such that you see what you've dropped. Since the CRC failed you're seeing garbage anyway.